book/src/design/gadgets/ecc/fixed-base-scalar-mul.md

```diff
@@ -174,7 +168,8 @@ $$
 \begin{array}{|c|l|}
 \hline
 \text{Degree} & \text{Constraint} \\\hline
-3 & q_\text{mul-fixed-short} \cdot \left(s \cdot P_y - P'_y\right) = 0 \\\hline
+3 & q_\texttt{mul\_fixed\_short} \cdot \left(P'_y - P_y\right) \cdot \left(P'_y + P_y\right) = 0 \\\hline
```

Comment:
I'm confused by this; I had been under the impression that the implementation constrains `s \in {-1, 1}`, which is simpler and should be equivalent.

Reply:
It is present in the doc further up; this book page splits the constraints across two boxes. https://github.com/zcash/halo2/blob/bf459804f66b322317f0cddb2d727a26378765df/book/src/design/gadgets/ecc/fixed-base-scalar-mul.md?plain=1#L100

I will have a follow-up PR adding protocol links for all of this 🙂

Reply:
Also, this constraint above is _not_ equivalent; if it were present but the `s^2 - 1` was not, then `s \in {-1, 0, 1}`.

Reply:
Yes, or m = 0 and P'<sub>y</sub> = P<sub>y</sub> = 0 in which case s could be anything. What I meant is that it's functionally equivalent given how s is used (it wouldn't matter if it were unconstrained in that case). It's academic though given that we do have the q<sub>`mul_fixed_short`</sub>.(s<sup>2</sup> - 1) = 0 constraint.

Reply:
It might be a good idea to add a link pointing up to the previously defined constraints, though.